File: BinarytoDecimal.rb
#!/usr/bin/env ruby
def BinarytoDecimal(binaryNumber)
	value = binaryNumber.to_s
	len = value.length
	output = 0
	for position in 0..len
		char = value[position]
		if char == "1"
			output += 2**(len - 1 - position)
		elsif char == "0"
			nil
		elsif value[position].to_i.between?(2, 9)
			return "Make sure you only input binary values"
		end
	end
	return output
end

print "Binary Number: "
num = gets.to_i
puts BinarytoDecimal(num)

File: BinarytoDecimal.pl
#!/bin/perl
use warnings;

sub BinarytoDecimal() {
  my $value = "@_";
  my $len = length($value);
  my $output = 0;
  for (my $position = 0; $position < $len; $position++) {
    my $char = substr($value, $position, 1);
    if ($char == "1") {
      $output += 2**($len - 1 - $position);
    } elsif ($char == "0") {
    
    } else {
      return "Make sure you only input binary values";
    }
  }
  return $output;
}

print("Binary Number: ");
$out = &BinarytoDecimal(int(<>));
print("$out\n");
